### PR TITLE
[torch.compile] register allreduce operations as custom ops

### PR DESCRIPTION
**Pull request by youkaichao**
rework of https://github.com/vllm-project/vllm/pull/7755

instead of embedding the `tp` name in the op name, embed the group name into op's argument. this can naturally extend to future ops like allreduce in context parallel.

### CODE REVIEW

**Comment by youkaichao** [comment]
@SageMoore can you help review this? thanks!

most of the code come from your pr, and i just reworked the python side registration code.

**Review by SageMoore — APPROVED** [review]
LGTM

**Review by robertgshaw2-redhat — APPROVED** [review]

File: vllm/distributed/parallel_state.py [hunk 1]
```diff
@@ -69,6 +70,58 @@ def _split_tensor_dict(
...
+def inplace_all_reduce(tensor: torch.Tensor, group_name: str) -> None:
+    assert group_name in _groups, f"Group {group_name} is not found."
+    group = _groups[group_name]()
+    if group is None:
+        raise ValueError(f"Group {group_name} is destroyed.")
+    group._all_reduce(tensor)
+
+
+@inplace_all_reduce.register_fake
+def _(tensor: torch.Tensor, group_name: str) -> None:
```
Comment (SageMoore) [inline vllm/distributed/parallel_state.py:107]: Nit: FWIW, I don't think you need this meta function

Comment (youkaichao) [reply vllm/distributed/parallel_state.py:107]: it is needed when we use inductor. if we use the `eager` backend, it is not needed.

**Comment by youkaichao** [comment]
merge as failing tests are not relevant

**Pull request merged**